packages/adminrouter/extra/src/README.md

```diff
@@ -870,3 +1112,842 @@ expressions/statements:
 This is why the context of each endpoint is not protected by locks, in
 case when it's only about fetching a single value from context dict or
 storing/appending one there.
+
+## Tests writing howto
+
+This section is intended to quickly enable contributors to start writing tests.
+All the test harness components and concepts are described here with a "bigger
+picture" in mind so that it is easier to understand how they relate to each
+other.
+
+It is divided into three sections:
+* generic tests which can be covered by simply updating test harness' YAML
+  configuration
+* standard tests that do not fit into YAML-described generic tests
+* advanced usage
+
+It is possible that some of the paragraphs will be repeating the information
+already provided earlier in the text, while some others may require reading or
+re-reading earlier paragraphs. It is very much recommended though to read all the
+preceding sections before reading this one.
```

Comment:
This is very meta and unless you are a highly trained master of prose I do not recommend instructing your readers like that :-). 

Recommendation: remove paragraph.